### PR TITLE
Added "node:events" to external config for the functionsmap generation. #244

### LINKED ISSUE #244: [🐛 Bug]: Could not map all functions to an entry in the middleware manifest
### next-on-pages environment related information

```
System:
        Platform: darwin
        Arch: arm64
        Version: Darwin Kernel Version 22.1.0: Sun Oct  9 20:15:09 PDT 2022; root:xnu-8792.41.9~2/RELEASE_ARM64_T6000
        CPU: (10) arm64 Apple M1 Max
        Memory: 32 GB
        Shell: /bin/zsh
Binaries:
        Node: 19.0.0
        Yarn: N/A
        npm: 8.19.2
        pnpm: N/A
Package Manager Used: npm
Relevant Packages:
        @cloudflare/next-on-pages: 0.10.1
        vercel: N/A
        next: N/A
```

### Description

When trying to build with `npx @cloudflare/next-on-pages --experimental-minify` I am getting the error: 
`⚡️ ERROR: Could not map all functions to an entry in the middleware manifest.`

The output folder looks fine to me too, however I suspect there is a worker.js file missing that enables the functions in the first place. The deployment via wrangler also shows the static assets, while the functions are missing completely.

### Reproduction

_No response_

### Pages Deployment Method

Direct Upload (`wrangler pages publish` or the @cloudflare/pages-action GitHub Action)

### Pages Deployment ID

_No response_

### Additional Information

Output log:
```
▲  Route (app)                                Size     First Load JS
▲  ┌ ℇ /                                      745 B           404 kB
▲  ├ ℇ /eula                                  1.17 kB        85.9 kB
▲  ├ ℇ /legal-notice                          1.17 kB        85.9 kB
▲  ├ ℇ /login                                 881 B           404 kB
▲  ├ ℇ /privacy-policy                        1.17 kB        85.9 kB
▲  ├ ℇ /settings                              167 B          77.5 kB
▲  ├ ℇ /settings                              167 B          77.5 kB
▲  ├ ℇ /settings                              167 B          77.5 kB
▲  ├ ℇ /settings/billing                      0 B                0 B
▲  ├ ℇ /settings/billing                      0 B                0 B
▲  ├ ℇ /settings/developer                    0 B                0 B
▲  ├ ℇ /settings/developer                    0 B                0 B
▲  ├ ℇ /settings/notification                 0 B                0 B
▲  ├ ℇ /settings/notification                 0 B                0 B
▲  ├ ℇ /settings/privacy                      0 B                0 B
▲  ├ ℇ /settings/privacy                      0 B                0 B
▲  ├ ℇ /settings/profile                      0 B                0 B
▲  ├ ℇ /settings/profile                      0 B                0 B
▲  ├ ℇ /settings/storage                      0 B                0 B
▲  ├ ℇ /settings/storage                      0 B                0 B
▲  ├ ℇ /settings/subscription                 0 B                0 B
▲  ├ ℇ /settings/subscription                 0 B                0 B
▲  ├ ℇ /settings/usage                        0 B                0 B
▲  └ ℇ /settings/usage                        0 B                0 B
▲  + First Load JS shared by all              77.3 kB
▲  ├ chunks/253-d9f504f8f620858c.js         24.5 kB
▲  ├ chunks/6b645fa8-5c940de39a313a0d.js    50.5 kB
▲  ├ chunks/main-app-b8b5439f916e9ad1.js    215 B
▲  └ chunks/webpack-f1d4f8cbe4f57b73.js     2.06 kB
▲  
▲  Route (pages)                              Size     First Load JS
▲  ─ ○ /404                                   179 B          86.5 kB
▲  + First Load JS shared by all              86.3 kB
▲  ├ chunks/main-1d27128c79c25202.js        84 kB
▲  ├ chunks/pages/_app-7fb87454d03f3d10.js  193 B
▲  └ chunks/webpack-f1d4f8cbe4f57b73.js     2.06 kB
▲  
▲  ƒ Middleware                               23.9 kB
▲  ℇ  (Streaming)  server-side renders with streaming (uses React 18 SSR streaming or Server Components)
▲  ○  (Static)     automatically rendered as static HTML (uses no initial props)
▲  Traced Next.js server files in: 13.862ms
▲  Created all serverless functions in: 16.737s
▲  Collected static files (public/, static/, .next/static): 9.708ms
▲  Using TypeScript 4.9.3 (local user-provided)
▲  Build Completed in .vercel/output [3m]
⚡️ Completed `npx vercel build`.
⚡️ ⚡️ ERROR: Could not map all functions to an entry in the middleware manifest.
⚡️ Please report this at https://github.com/cloudflare/next-on-pages/issues.
```
I am trying to use the new parallel rendering feature.

### Would you like to help?

- [X] Would you like to help fixing this bug?

### PR DESCRIPTION
**Pull request by felix-schultz**
resolves #244

### CODE REVIEW

**Comment by felix-schultz** [comment]
#244

Added `node:events` to external. 

- TODO: handle runtime usage (this version might lead to runtime crashes?)

**Review by dario-piotrowicz — COMMENTED** [review]
Thanks a lot @felix-schultz 😄

Sorry this was by bad for not excluding all node modules here

In the future we might need some additional esbuild plugin as @james-elicx mentioned in your issue, but anyways I think we should just merge this as is for now as it is anyways improving the situation and aligning it with [this](https://github.com/cloudflare/next-on-pages/blob/0d9e155710fb2dea31c6692ef2bebbc54f5a879e/src/buildApplication/buildWorkerFile.ts#LL71C13-L71C23)

If/when we'll encounter issues in which we need the plugin we'll deal with that then 🙂

(also if user code imports `node:events` via dynamic imports the code before your change would break, but not after, so it's still a win for us I think 🙂)

__

Please check out my suggestions, also please add a `patch` changeset by running `npm run changeset` 🙏

File: src/buildApplication/generateFunctionsMap.ts [hunk 1]
```diff
@@ -399,7 +399,7 @@ async function buildWebpackChunkFiles(
 			platform: 'neutral',
 			outfile: chunkFilePath,
 			bundle: true,
-			external: ['node:buffer'],
+			external: ['node:buffer', 'node:events'],
```
[the remark this answers is not in the document]
Comment (felix-schultz) [reply src/buildApplication/generateFunctionsMap.ts]: Done 👍🏻

**Review by dario-piotrowicz — APPROVED** [review]
@felix-schultz awesome! thanks a lot! 😃👍

**Pull request merged**